packages/runner/src/main.jsx

```diff
@@ -18,11 +18,35 @@ const Runner = {
 
       const NO_COMMAND_LOG = config.env && config.env.NO_COMMAND_LOG
 
-      const state = new State(NO_COMMAND_LOG ? 0 : (config.state || {}).reporterWidth)
+      const state = new State({
+        reporterWidth: NO_COMMAND_LOG ? 0 : (config.state || {}).reporterWidth,
+        specs: config.specs,
+        specListWidth: NO_COMMAND_LOG ? 0 : SPEC_LIST_WIDTH,
+      })
 
       Runner.state = state
       Runner.configureMobx = configure
 
+      const setSpecByUrlHash = () => {
+        const specPath = util.integrationSpecPath()
+
+        if (specPath) {
+          state.updateSpecByUrl(specPath)
+        }
+      }
+
+      setSpecByUrlHash()
+
+      autorun(() => {
```

Comment:
This is how we change the hash and run the test you selected in the spec list. Stolen from https://github.com/cypress-io/cypress/blob/develop/packages/runner-ct/src/main.jsx#L42-L61